src/error.rs

```diff
@@ -3,8 +3,21 @@
 
 use thiserror::Error;
 
-pub type Result<T, E = Error> = std::result::Result<T, E>;
+#[cfg(not(feature = "std"))]
+extern crate alloc;
 
+#[cfg(not(feature = "std"))]
+use alloc::boxed::Box;
+
+#[cfg(feature = "std")]
+use std::error::Error as StdError;
+
+#[cfg(not(feature = "std"))]
+use core::error::Error as CoreError;
```

Comment:
```suggestion
use core::error::Error as CoreError;
```

`std::error::Error` is a reexport of `core::error::Error` so you don't need to differentiate between them, you can simplify this to always using the core version. Similarly, you can always use `alloc::string::String` and `alloc::boxed::Box` and it will be the same types as the std versions if the std library is available.

So in the end, your own `Error` enum does not even need feature-gated duplication.